### PR TITLE
✨Improve Welcome Page

### LINKED ISSUE #942: ✨ Improve Welcome Page; improve UX
## Describe your issue

Zur Verbesserung der UX kann der Einstieg für den User verbessert werden.

## Possible solution

VS-Code macht so etwas:

screen shot 2018-09-14 at 17 29 47

Das finde ich gut und es wäre eine Lösung für den Anfang; aber maximal für Version 1.

Endausbau, sollte so etwas sein:

screen shot 2018-09-17 at 17 34 21

Das ist die Welcome-Seite von Code.

Code hat ein anderes Großartiges Feature:

screen shot 2018-09-18 at 10 09 46

Das ist ein Overlay, dass die Feature zeigt.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### LINKED ISSUE #1076: 🚸 Improve Opening Folders/Files
#### Describe your issue

We should make the opening of folders and files more intuitive.
Maybe add a 'Open Folder'/'Open File' text that can be used, like in the welcome page in VSCode (also have a look at #942).

Also we should maybe add the 'File' Section to the Toolbar, like most of the text editors have.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by Vyperus**
**Changes:**

1. Add Open Solution Shortcut to Welcome Page.
2. Add Open File Shortcut to Welcome Page.
3. Add Create Diagram Functionality out of Welcome Page. 
(This will default to the "Desktop" but if you select a Solution the diagram will be created there instead.)
4. Add Create new Diagram Shortcut to Welcome Page.

bildschirmfoto 2019-01-14 um 16 52 37

**Issues:**

Closes #942 
Closes #1076

PR: #PullRequest

## How can others test the changes?

Go through the shortcuts now mentioned on the start-page.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

**Comment by cmg-dev** [comment]
Why don't we copy the layout of VS Code?! Please copy the layout...

File: src/modules/start-page/start-page.html [hunk 1]
```diff
@@ -4,6 +4,23 @@
...
+        </div>
+        <div class="quick-start__item">
+          <div class="quick-start__topic" click.delegate="openLocalSolution()">Open Solution</div>
+          <div show.bind="isRunningInElectron && isRunningOnMacOS" class="quick-start__shortcut">⌘ + ⇧ + O</div>
+          <div show.bind="isRunningInElectron && (isRunningOnWindows || isRunningOnLinux)" class="quick-start__shortcut">Strg + Shift + O</div>
+        </div>
+        <div class="quick-start__item">
+          <div class="quick-start__topic" click.delegate="createNewSingleDiagram()">New Diagram</div>
+          <div show.bind="isRunningInElectron && isRunningOnMacOS" class="quick-start__shortcut">⌘ + N</div>
+          <div show.bind="isRunningInElectron && (isRunningOnWindows || isRunningOnLinux)" class="quick-start__shortcut">Strg + N</div>
```
Comment (cmg-dev) [inline src/modules/start-page/start-page.html]: Just make three lines. one for each OS.

File: src/modules/start-page/start-page.ts [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+import environment from '../../environment';
+
+@inject(EventAggregator)
 export class StartPage {
+  private _eventAggregator: EventAggregator;
+
+  public isRunningInElectron: boolean = (window as any).nodeRequire;
+  public isRunningOnWindows: boolean;
+  public isRunningOnMacOS: boolean;
+  public isRunningOnLinux: boolean;
```
[the remark this answers is not in the document]
Comment (cmg-dev) [reply src/modules/start-page/start-page.ts]: @LeoTT 

Good find!

[the remark this answers is not in the document]
Comment (cmg-dev) [reply src/modules/start-page/start-page.ts]: I support this.

**Comment by alexanderkasten** [comment]
> (This will default to the "Desktop" but if you select a Solution the diagram will be created there instead.)

How to select a solution 🤔 ?

File: src/modules/start-page/start-page.html [hunk 3]
```diff
@@ -4,6 +4,26 @@
...
+      <div class="start-page__quick-start" show.bind="isRunningInElectron">
+        <div class="quick-start__item">
+          <div class="quick-start__topic" click.delegate="openSingleDiagram()">Open Diagram</div>
+          <div show.bind="isRunningInElectron && isRunningOnMacOS" class="quick-start__shortcut">⌘ + O</div>
+          <div show.bind="isRunningInElectron && isRunningOnWindows" class="quick-start__shortcut">Strg + O</div>
+          <div show.bind="isRunningInElectron && isRunningOnLinux" class="quick-start__shortcut">Strg + O</div>
+        </div>
+        <div class="quick-start__item">
+          <div class="quick-start__topic" click.delegate="openLocalSolution()">Open Solution</div>
+          <div show.bind="isRunningInElectron && isRunningOnMacOS" class="quick-start__shortcut">⌘ + ⇧ + O</div>
```
Comment (alexanderkasten) [inline src/modules/start-page/start-page.html]: Would make the shortcuts not dragable like the image or the welcome text

File: src/modules/start-page/start-page.scss [hunk 4]
```diff
@@ -23,3 +22,39 @@
...
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+}
+
+.quick-start__topic {
+  margin-right: 20px;
+  color: #555555;
+  font-size: 17px;
+  font-weight: bold;
```
Comment (alexanderkasten) [inline src/modules/start-page/start-page.scss]: IMO font-size 17 and also font-weight bold is a bit too much, would remove one of them

File: src/modules/start-page/start-page.html (hunk 3 above)
Comment (Vyperus) [reply src/modules/start-page/start-page.html]: The "__content" is not dragable. Shouldn't the child elements of it non dragable as well?

**Comment by Vyperus** [comment]
> > (This will default to the "Desktop" but if you select a Solution the diagram will be created there instead.)
> 
> How to select a solution 🤔 ?

Right now you have to open a solution and open a diagram in it.

File: src/modules/start-page/start-page.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (robpal94) [reply src/modules/start-page/start-page.ts]: We should also apply this rule for other types, such as `numbers` or `strings`.

**Comment by SteffenKn** [comment]
Simply add 'user-select: none;' to '.start-page__content'

**Pull request merged**